src/plugin_host.jl

```diff
@@ -0,0 +1,174 @@
+start_host() = try wait(nvim_child(HostHandler()).reader) end
+
+immutable HostHandler
+    loaded_plugins::Set{String} # plugin file paths
+    proc_callbacks::Dict{String, Function}
+end
+
+function HostHandler()
+    HostHandler(Set{String}(), Dict{String, Function}())
+end
+
+# names/methods for invoked cmds/fns have the form:
+#   <file path>:(command|function|autocmd):<procedure name>
+function on_notify(h::HostHandler, c, name::String, args::Vector{Any})
+    proc = require_plug(h, name)
+
+    if proc == nothing
+        throw(ErrorException(name * " not defined"))
+    end
+
+    @async(try
+        proc(c, args...)
+    catch
+        throw(MethodError(name, tuple(NvimClient, args...)))
+        throw(err)
```

Comment:
`err` is not defined here. But I don't really think this is a context we want to throw anything, rather let's just `println(STDERR)` what we want to have here. Perhaps there is some function to print the exception with traceback.